pkg/epp/scheduling/plugins/scorer/weighted_scorer.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scorer
+
+import (
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/plugins"
+)
+
+// NewWeightedScorer initializes a new WeightedScorer and returns its pointer.
+func NewWeightedScorer(scorer plugins.Scorer, weight int) *WeightedScorer {
```

Comment:
I think all scorers should have weights, and so why do we need this special type?

Reply:
I guess when we define the config api, we will have a Plugin type that would include a weight parameter.

Reply:
If we introduce this WeighedScorer, then scheduler can just maintain a list of WeightedScorers instead of a `map[Scorer]int`.

Reply:
good idea 👍